src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        initDuke();
+        displayWelcome();
+        while (shouldContinue) {
+            readInput();
+            setAction();
+            processCommand();
+        }
+    }
+
+    private static void processCommand() {
+        switch(action) {
+        case "bye":
+            sayBye();
+            break;
+        case "list":
+            listTasks();
+            break;
+        case "done":
+            completeTask();
+            break;
+        case "deadline":
+            setDeadline();
+            confirmTask();
+            break;
+        case "event":
+            createEvent();
+            confirmTask();
+            break;
+        case "todo":
+            addTodo();
+            confirmTask();
+            break;
+        }
+    }
+
+    private static void addTodo() {
+        String description = command.substring(5);
+        instructions[count] = new Todo(description);
+        count += 1;
+    }
+
+    private static void createEvent() {
+        int index = command.indexOf("/at");
+        String description = command.substring(6, index - 1);
+        String duration = command.substring(index + 4);
+        instructions[count] = new Event(description, duration);
+        count += 1;
+    }
+
+    private static void setDeadline() {
+        int index = command.indexOf("/by");
+        String description = command.substring(9, index - 1);
+        String by = command.substring(index + 4);
+        instructions[count] = new Deadline(description, by);
+        count += 1;
+    }
+    private static void completeTask() {
+        String[] phrases = command.split(" ");
+        int index = Integer.parseInt(phrases[1]);
+        instructions[index-1].markAsDone();
+        System.out.println(completeMessage);
+        System.out.println("  " + instructions[index-1]);
+    }
+
+    private static void setAction() {
+        String[] phrases = command.split(" ");
+        action = phrases[0];
+    }
+
+    private static void listTasks() {
+        System.out.println(listMessage);
+        for (int i = 0; i < count; i++) {
+            System.out.println((i+1) + "." + instructions[i]);
+        }
+    }
+
+    private static void sayBye() {
+        System.out.println(byeMessage);
+        shouldContinue = false;
+    }
+
+    private static void readInput() {
+        command = input.nextLine();
+    }
+
+    private static void initDuke() {
+        shouldContinue = true;
+        input = new Scanner(System.in);
+        instructions = new Task[MAXIMUM_TASKS];
+        count = 0;
+    }
+
+    private static void confirmTask() {
+        System.out.println("Got it. I've added this task:\n  " + instructions[count-1] + "\n"
+            + "Now you have " + count + " tasks in the list.");
+    }
```

Comment:
Consider using String.format or refracting this line to make it clearer to the user.

Reply:
Ah, I didn't think of that. Noted!